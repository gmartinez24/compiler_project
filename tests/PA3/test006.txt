// Grant Martinez
// test printing return of function call

main

int a;
int b;

function maxInt (int x, int y) : int
{
    if (x > y) then
        return x;
    else
        return y;
    fi;
};


{
    a = call readInt(); // 20
    b = call readInt(); // 63
    call println();
    call printInt(call maxInt(a,b)); // should be 63

}.