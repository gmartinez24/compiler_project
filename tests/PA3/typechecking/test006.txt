// array accessing

main

int[10] a;
int x, y;

function foo(int x, int y) : bool
{
    return x < y;
};

function bar(int x, int y) : int
{
    return x + y;
};

{
    x = 0;
    y = 10;

    call printInt( a[-5] ); // error: indexing with negative int constant
    call printInt( a[-5] ); // error: indexing with negative int constant
    call printInt( a[call bar(x,y)] ); // ok
    call printInt( a[call foo(x,y)] ); // error: indexing with bool
    call printInt( a[x + call foo(x,y)] ); // error: int + bool
    call printInt( a[x * y % a[x]] ); // ok
    call printInt( a[call foo(x,y) + call bar(x,y)] ); // error: int + bool
    

}.