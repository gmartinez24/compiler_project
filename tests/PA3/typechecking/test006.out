Error type-checking file.
TypeError(22,26)[Array Index Out of Bounds : -5 for array a]
TypeError(22,20)[Cannot dereference ErrorType(Array Index Out of Bounds : -5 for array a)]
TypeError(22,5)[Call with args TypeList(ErrorType(Cannot dereference ErrorType(Array Index Out of Bounds : -5 for array a))) matches no function signature.]
TypeError(23,26)[Array Index Out of Bounds : -5 for array a]
TypeError(23,20)[Cannot dereference ErrorType(Array Index Out of Bounds : -5 for array a)]
TypeError(23,5)[Call with args TypeList(ErrorType(Cannot dereference ErrorType(Array Index Out of Bounds : -5 for array a))) matches no function signature.]
TypeError(25,37)[Cannot index int[10] with bool.]
TypeError(25,20)[Cannot dereference ErrorType(Cannot index int[10] with bool.)]
TypeError(25,5)[Call with args TypeList(ErrorType(Cannot dereference ErrorType(Cannot index int[10] with bool.))) matches no function signature.]
TypeError(26,24)[Cannot add int to bool.]
TypeError(26,41)[Cannot index int[10] with ErrorType(Cannot add int to bool.).]
TypeError(26,20)[Cannot dereference ErrorType(Cannot index int[10] with ErrorType(Cannot add int to bool.).)]
TypeError(26,5)[Call with args TypeList(ErrorType(Cannot dereference ErrorType(Cannot index int[10] with ErrorType(Cannot add int to bool.).))) matches no function signature.]
TypeError(28,36)[Cannot add bool to int.]
TypeError(28,53)[Cannot index int[10] with ErrorType(Cannot add bool to int.).]
TypeError(28,20)[Cannot dereference ErrorType(Cannot index int[10] with ErrorType(Cannot add bool to int.).)]
TypeError(28,5)[Call with args TypeList(ErrorType(Cannot dereference ErrorType(Cannot index int[10] with ErrorType(Cannot add bool to int.).))) matches no function signature.]

